cql3/statements/select_statement.cc

```diff
@@ -1118,12 +1118,15 @@ query::partition_slice indexed_table_select_statement::get_partition_slice_for_g
     partition_slice_builder partition_slice_builder{*_view_schema};
 
     if (!_restrictions->has_partition_key_unrestricted_components()) {
+        // Computed token column to be added to index view restrictions
+        const column_definition& token_cdef = *_view_schema->clustering_key_columns().begin();
+        // Is secondary index with new token_column_computation type (computation name: token_v2)
+        const bool is_token_v2_index = token_cdef.type == long_type;
+
         auto single_pk_restrictions = dynamic_pointer_cast<restrictions::single_column_partition_key_restrictions>(_restrictions->get_partition_key_restrictions());
-        // Only EQ restrictions on base partition key can be used in an index view query
         if (single_pk_restrictions && single_pk_restrictions->is_all_eq()) {
+            // EQ restrictions on base partition key
             auto clustering_restrictions = ::make_shared<restrictions::single_column_clustering_key_restrictions>(_view_schema, *single_pk_restrictions);
-            // Computed token column needs to be added to index view restrictions
-            const column_definition& token_cdef = *_view_schema->clustering_key_columns().begin();
```

Comment:
I don't understand how you can drop this line, you need token_cdef a few lines down... What am I missing?

Reply:
I moved it a few lines up, outside `if`, because it is now also used in `else` case.

Reply:
Now my previous comment is out-of-date, as `token_cdef` is moved into `_global_index_prepare_data`.